tcl/return.tcl

```diff
@@ -3,54 +3,68 @@ namespace eval qc {
 }
 
 proc qc::return2client { args } {
-    #| Return data to http client
-    # Usage return2client ?code code? ?content-type mime-type? ?file file? ?filename filename? ?html html? ?text text? ?xml xml? ?json json? ?filter_cc boolean? ?header header? .. 
+    #| Return data to http client if a connection exists otherwise just output the given content.
+    # Usage return2client ?code code? ?content-type mime-type? ?html html? ?text text? ?xml xml? ?json json? ?file file? ?filename filename? ?download boolean? ?filter_cc boolean? ?header header? .. 
     set arg_names [qc::args2vars $args]
-    set headers [lexclude $arg_names file filename html xml text json code content-type]
+    set headers [lexclude $arg_names file filename html xml text json file filename download code content-type filter_cc]
     default code 200
     default filter_cc no
+    default filename [qc::url_path [ns_conn url]]
```

Comment:
Have you tested this when no connection exists? I would expect it to error if called from a cp or tclsh